src/modules/core/components/DetailsWidgetUser/DetailsWidgetUser.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+import HookedUserAvatar from '~users/HookedUserAvatar';
+import MaskedAddress from '~core/MaskedAddress';
+import InvisibleCopyableAddress from '~core/InvisibleCopyableAddress';
+import { useUser } from '~data/index';
+import { Address } from '~types/index';
+
+import styles from './DetailsWidgetUser.css';
+
+const displayName = 'DetailsWidgetUser';
+
+interface Props {
+  walletAddress: Address;
+}
+
+const DetailsWidgetUser = ({ walletAddress }: Props) => {
+  const UserAvatar = HookedUserAvatar({ fetchUser: false });
+  const userProfile = useUser(walletAddress);
+  const username = userProfile?.profile?.username;
+
+  return (
+    <div className={styles.main}>
+      <UserAvatar
+        size="s"
+        notSet={false}
+        address={walletAddress || ''}
+        showInfo
+      />
+      {username && <div className={styles.username}>@{username}</div>}
```

Comment:
This needs to display the user's display name as well, so something like:
```suggestion
      {username && <div className={styles.username}>{displayName || `@${username}`}</div>}
```